---
title: Seeds of Success - Diversity and impact of collections conserved in the USDA
  National Plant Germplasm System
author: Daniel Carver, Colin Khoury, Stephanie Greene, (potentially additional authors
  from Pullman, BLM)
date: "June 22, 2018"
bibliography: sosReportBib.bib
output:
  html_document:
    code_folding: hide
    df_print: paged
    highlight: tango
    number_sections: yes
    theme: yeti
    toc: yes
    toc_depth: 2
    toc_float:
      collapsed: no
      smooth_scroll: no
---


```{r, echo=FALSE, message=FALSE, warning=FALSE}
#import all necessary libraries
if (!require('dplyr')) install.packages('dplyr')
if (!require('ggplot2')) install.packages('ggplot2')
if (!require('plyr')) install.packages('plyr')
if (!require('tidyr')) install.packages('tidyr')
if (!require('tm')) install.packages('tm')
if (!require('SnowballC')) install.packages('SnowballC')
if (!require('wordcloud')) install.packages('wordcloud')
if (!require('DT')) install.packages('DT')


library(knitr)
library(dplyr)
library(ggplot2)
library(plyr)
library(tidyr)
library(tm)
library(SnowballC)
library(wordcloud)
library(DT)
#install.packages('devtools')
#library(devtools)
#install_version("rmarkdown",version=1.8)

```

**DRAFT**


- this is a working document

- at this stage the majority of the effort of the authors has been placed into developing a series of graphic/tables that alllows an exploration of the data (see results).

- The text within the document is preliminary and is expected to undergo major revisions based in part on input from outside contributors, as such much of the written content is merely an outline of what is to be expected.

- (from Colin- we need to provide a report to BLM. But for the academic paper, I personally think we should collaborate with BLM to talk about use in general including for restoration. This would be a much stronger paper, because overall the CWR and other useful plants aren't that many)

# Abstract {-}

Summarize the relationship between the BLM Seeds Of Success (SOS) program and the USDA ARS NPGS conservation efforts. These two organizations are collaborating on collecting, conserving, and making available for use a wide diversity of native plants occurring on public lands.

# Introduction
```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
base_path <- "C:/Users/Daniel.Carver/Documents/newH/SOS_Project"

#import SOS data
sosData <- read.csv(paste0(base_path, "/analysisData/SOSdata_20180614_forARS.csv")) 
#Find the total number of recorded entries
totalSamples <- nrow(sosData)


##import K13 data
K13Data <- read.csv(paste0(base_path, "/analysisData/CWR_US_inventory_2013_7_10.csv"))

  
#import WEP data
wepData <- read.csv(paste0(base_path, "/analysisData/Distributions&NativityOfSpeciesWithEconomicUses.csv"))
wepUse <- read.csv(paste0(base_path, "/analysisData/EconomicUseCategoriesOfSpecies.csv"))

#import distribution data GRIN 
allGrin <- read.csv(paste0(base_path, "/analysisData/sosAllAccessions.csv"))
newOrders <- read.csv(paste0(base_path, "/analysisData/allSOSOrders20181015.csv"))


#import germination data 
viability <- read.csv(paste0(base_path, "/analysisData/generaAverageGermination.csv"))


```



The Seeds of Success (SOS) program was established in 2001 to facilitate the collection of native seeds. To make use of the collected materials, the program partners with land management agencies to support the re-vegetation of disturbed or degraded areas, academic and professional partners to promote and advance scientific research, and the United States Department of Agriculture (USDA) for the long term preservation of species genetic diversity.

The Seeds of Success program provides the boots on the ground effort in the collection of native seed, particularly for restoration. The program partners with the Agricultural Research Service of the USDA to store, maintain, and distribute the seeds collected. The Pullman, Washington section of the ARS is the group responsible for the initial on boarding and entering of the seeds into the Genetic Resource Inventory Network (GRIN) as well as maintaining the seeds in environments that support preservation.

This article provides an overview of the diversity of plants collected by the SOS program. It then examines the conservation of this diversity within the USDA Agricultural Research Service National Plant Germplasm System (NPGS) and the distribution of samples from the NPGS to researchers. Particular focus is given to crop wild relatives which are useful as genetic resources for the improvement of cultivated plants, and other species which have been used as sources of food, medicine, forage, and other purposes.


# Methods

- data acquisition
  - SOS BLM: Information regarding the SOS collection was provided as a csv file by the Seeds of Success Program National Collection Curator in June of 2018. This dataset is continually updated as more field collecting occurs. All data from the years 2017 and earlier is assumed to be correct and complete. The data collected by the SOS program contains a rich set of specific information regarding the plant from which seed is collected as well as the geographic provenance and habitat that the plant was collected from. This dataset is made available by request only and can only be used with the express permission of the BLM SOS program.

  - GRIN Global : The NPGS's Germplasm Resource Information Network(GRIN-Global) provides public access to informationa regardin accessions maintained by the System. The data is publicly available online. GRIN Global data for all SOS accessions were retrieved from the database by selecting all accessions that have an action_name_code of "SOS" and stored as csvs.
  - USDA ARS Pullman office: Before assessing the datasets a meeting of individuals involved with the management of the SOS accession was held to identify the primary questions, define the process of transferring the accessions between the agencies, and understand the processing and management of the collections once they enter the NPGS.


- data processing
  - rmarkdown : The datasets were processed, joined, and queried using the statistical programming language R. Specific libraries that were used include: plyr, tidyr, dplyr, and ggplot2.  The report is compiled as an RMarkdown document to help with the reproducibility of the content as the relationship between the two organizations continues to develop.


# Results

The results are split into sections based on the primary questions we attempted to answer. We should probably highlight the questions earily in the document but for now ther are listed here.

1.  How diverse is the SOS collection?
2.  How many of the SOS accessions can be identified as crop wild realtives (CWR), wild utilized species (WUS), and/or economically valuable plants (i.e. listed in GRIN World Economic Plants)?
3.  What is the level of representation of SOS collections within the NPGS?
4.  Where are SOS accessions curated within the NPGS?
5.  What is the state of SOS collections in NPGS with regard to viability?
6.  Who requests SOS accessions from NPGS and for what purposes?





```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE , caption="Number of Collections per Family from the SOS program." }
# histogram with added parameters
ggplot(sosData, aes(COLL_YR))+
  geom_histogram(binwidth = 1,
                 col="white",
                 fill="blue",
                 alpha = 0.5) +
  labs(title = "Number of Collections")+
  labs(x = "Collection Year", y= "Number of Accessions") +
  xlim(c(2000,2018))+
  ylim(c(0,3500))+
  scale_x_continuous(breaks = seq(2000,2018, 2))+
  scale_y_continuous(breaks = seq(0,3500, 500))+
  theme_minimal()

```
<br>
<br>


The figure above illustrates the number of accessions collected by the SOS program over time. At this point in time the 2018 sample have not been added to the SOS databased and therefore are not represented on this figure.


## Diversity

```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}

### TODO
### write a function that takes dataset and column and returns the uniquenubmer of accesssions

#Determine the total number of unique families collect and the number of collection in each family
uniqueFam <- dplyr::summarise(sosData,n_distinct(sosData$FAMILY))
uniqueFam1 <- dplyr::count(sosData,FAMILY) #keep this in hear incase we want to display it

## Determine the total number of unique GENUS collect and the number of collection in each GENUS
uniqueGen <-  dplyr::summarise(sosData,n_distinct(sosData$GENUS))
uniqueGen1 <- dplyr::count(sosData, GENUS)

## Determine the total number of unique SPECIES collect and the number of collection in each SPECIES
uniqueSpe <-  dplyr::summarise(sosData,n_distinct(sosData$SPECIES))
uniqueSpe1 <- dplyr::count(sosData, SPECIES)

## Determine the total number of unique Taxa collect and the number of collection in each Taxa
uniqueTaxa <- summarise(sosData,n_distinct(sosData$NAME))
uniqueTaxa1 <- dplyr::count(sosData, NAME)

## DEfeine row and column names
labelsRow1 <- c("SOS Collection")
labelsCol1 <- c('All Accessions', "Total Number of Families", "Total Number of Genera", "Total Number of Taxa")

#Initiate an empty dataframe
sosSummary <- data.frame()

# create the first row of the dataframe
sosDataSum <- c(totalSamples, uniqueFam, uniqueGen, uniqueTaxa)

#bind the two df together
sosSummary <- tbl_df(rbind(sosSummary, sosDataSum))
#define column names and row names
colnames(sosSummary) <- labelsCol1
rownames(sosSummary) <- labelsRow1
sosSummary
```




This Seeds of Success program since its inception has collected a total of `r totalSamples` accessions. These are comprised of `r uniqueTaxa[1,]` unique taxa from `r uniqueGen[1,]` genera from `r uniqueFam[1,]` plant families.



The [Flora of North America Association](http://floranorthamerica.org/Outreach/science_outreach) provides a published reference source that provides a comrehesive and systematic account of the plant species in North American North of Mexico.

> Species descriptions are written and reviewed by experts from the systematic botanical community worldwide, based on original observations of living and herbarium specimens supplemented by a crucial review of the literature. Each treatment includes scientific and common names, taxonomic descriptions, identification keys, distribution maps, illustrations, summaries of habitat and geographic ranges, pertinent synonomy, chromosome numbers, phenology, ethnobotanical uses and toxicity, and other relevant biological information.
[FNA Introduction](http://floranorthamerica.org/introduction)

As of 2018, The FNA over has idenitified over 20,100 plant species that are native or naturalizes to North America north of Mexico, this includes both vascular and bryophytes. These species fall into 268 unique familes. This implies that the SOS program has collect seed from `r signif(100* (uniqueFam[1,] / 268),3)` percent of all unique familes and  approximately `r signif(100* (uniqueSpe[1,] / 20100),3) ` of all plant species native to the US and Canada.  

An accession is a sample of seeds from a single sampling location and in a single point in time. Collecting is performed, as possible, with the aim of harvesting enough seeds for short term storage, long term preservation, and for distribution.



```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, caption="Number of Collections per Family from the SOS program.", rows.print=10}
families <- dplyr::arrange(uniqueFam1, desc(n))
colnames(families) <- c("Family", "Total Number of Accessions")
DT::datatable(families)

```

Out of the `r uniqueFam[1,]` families collected half of the total accessions collected come from just 5 families. These are among the most widespread and diverse plant families in the USA.


```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, caption="Number of Collections per Family from the SOS program.", rows.print=10}
genera1 <- dplyr::arrange(uniqueGen1, desc(n))
colnames(genera1) <- c("Genera", "Total Number of Accessions")
DT::datatable(genera1)

```

This table shows the number of accession assocatied with the genera of all samples within the SOS data .

```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, caption="Number of Collections per Family from the SOS program.", rows.print=10}
taxa1 <- dplyr::arrange(uniqueTaxa1, desc(n))
colnames(taxa1) <- c("Taxa", "Total Number of Accessions")
DT::datatable(taxa1)

```

Table listing the number of collection for each speices. Most collected species are listed first. 


## Crop wild relatives, wild utilized species, and economic plants


Crop wild relatives are define in Maxted et al 2006 as

> A crop wild relative is a wild plant taxon that has an indirect use derived from
its relatively close genetic relationship to a crop; this relationship is defined in
terms of the CWR belonging to Gene Pools 1 or 2, or taxon groups 1 to 4 of the
crop.

This defination is most commonly applied to species that support are food and agricultural systems it can be extened to incorporate species that are harvested for economic or cultural purposes such as ornamentals, medicial uses, and foresty. (Maxted et al 2006). These CWR contain a genictic diversity which has been used to improvepest and desease resistance, tolerance to abioticstress, inceased yield, novel cytoplasms, and quality traits to multiple major crop species.(Khoury et al 2013) Yet the limiting factor in the utilizaiton of CWR is the collection, preservation, and accessability of genetic information for research scientist (Harlan, 1976). While this statement was made in 1976 recent work has illistrated that is still a primary matter of concern. (FAO, 2010: Fowler and Hodgekin, 2004).

A recent inventory of U.S. crop wild relatives (Khoury et al. 2013) recognized 2500 taxa from 160 genera and 56 plant families. Of these, around 300 native species were prioritized as potentially highly useful relatives of important food crops. These include the wild cousins of crops that were domesticated by indigenous peoples in the United States, such as sunflower and squash, as well as more recent domesticates, such as blueberry, cranberry, raspberry, blackberry, and pecan. Other priority species include the northernmost relatives of Mesoamerican crops such as corn, bean, chili pepper, and cotton, as well as relatives of temperate crops first domesticated in Europe and Asia, such as onion, grape, apple, hops and walnut. If properly collected, conserved, and made accessible to crop scientists, these crop wild relatives will constitute an extremely diverse portfolio supporting present and future crop improvement efforts.


We utilized the K13 dataset because is was generated to be the "a primary step in the the process toward a national stategy for the conservation of CWRs". The data sets makes an important distinstion between CWR and Wild Utilized Species(WUS. CWR are considered to be species that can directly be used for crop breeding. WUS are define by those that can be used for food, forage, medicine, herb, ornamental, and/or environmental restoration purposes. This extra level of distinction provides an import distinction regarding the intented use of the species. This dataset also provides a priority ranking system while allows for the identificantion of which species could have the greatest potential of crop genetic research. These details make this an ideal dataset for the interpretation of the true effect the SOS collections are having on the preservation of genetic resources.   


The World Econmic Plants dataset is derived from the second edition of World Economic Plants: A Standard Reference (Wiersema, J., Leï¿½n, B., 2013.). This book is acts as a catalog of economic plants, maintaing a standard scientific names, common names, economic impacts, and geographic distribution. The book contains 312 families and 3,585 genera. Economic impacts are split into 16 primary classes and 113 subclasses. It is important to not that of the 16 primary classes 3 are mostly negitive in there ecomonic impacts. These economic classes have been incorperate by the [GRIN economic plant database](https://npgsweb.ars-grin.gov/gringlobal/taxon/taxonomysearcheco.aspx).


This study is using dataset complied by Khoury et al 2013 and Wiersema and Leon 2013 to identify species that are considered crop wild relatives (CWR) and/or wild utilized species (WUS) (Khoury et al. 2013), or economic plants (WEP). The chart below is defining the primary acronyms that are used in the remainder of this document

```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
#aycromn referecne dataframe
#acronymDF <- data.frame()

# create the first column of the dataframe
abbrevNames <- c("SOS", "GRIN", "NLGRP", "WUS", "K13", "CWR", "WEP")
#create the meaning of acronym
nameMeaninng <- c("Seeds of Success", "Germplasm Resource Information Network", "National Laboratory Genetic Resource Presevation", "Wild Utilized Species", "Khoury et al 2013", "Crop Wild Relative", "Economic Plant")

#define column names and row names
acronymDF <-data_frame("Acronym" = abbrevNames, "Meaning" = nameMeaninng)
acronymDF

```




```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
###
# Create a summary chart showing the total number of accession, total taxa, # of CWR NA, # eco NA, # of economic WEP
# this may be added to also include the data about what's managed by the usda and what's at the NLGRP
#

# Replace the name column of the sos data so that the "ssp." notation is replaced with "subsp." as defined
# by the grin Data base
#import SOS data
# created a new column that has the same col name as the K13 data set. I will use the col to join on
sosData$Taxon <- sosData$NAME

#detemine the total number of taxon in dataset
sosTotalUnique <- n_distinct(sosData$Taxon)

#detemine the total number of taxon in dataset
K13TotalUnique <- K13Data %>% n_distinct("Taxon")

# join the data sets and trim the data so that taxon that do not have values for the K13 specific data columns are dropped.
trimCombinedData <- inner_join(sosData, K13Data, by='Taxon', type='left', match='all')

typeSum <- trimCombinedData %>%
  group_by(Type) %>%
  dplyr::summarise(uniqueNumber = n_distinct(Taxon), percentType = signif(100*(n_distinct(Taxon) / sosTotalUnique),3), totalAccession = n())


### read in WEP location data and filter to united states
wepLocal <- wepData %>%
  filter(country_code == 'USA') %>%
  dplyr::rename(Taxon = name) %>%
  distinct(Taxon)

###
# read in the WEP data and Rename the name column to match the newly created column from the SOS data
wepData1 <- wepUse %>%
  dplyr::rename(Taxon = name)


#Unique Join for species in the USA
### the WEP data has repeats of features if they have more then 1 economic use. I'm joining again here so the taxa are only represented once.

wepJoin <- inner_join(sosData, wepLocal, by='Taxon')

# join with sos data
combinedData2a <-  wepJoin %>%
  dplyr::summarise(uniqueNumber = n_distinct(Taxon), percentType = signif(100*(n_distinct(Taxon) / sosTotalUnique),3),  totalAccession = n())%>%
  mutate(Type = "wep")

endDf <- bind_rows(typeSum, combinedData2a)

#calculate total number of species from each dataset
uniqueWEP <- endDf[[3,2]]
uniqueWUS <- endDf[[2,2]]
uniqueCWR <- endDf[[1,2]]
#percent of sos in this class per dataset
percentWEP<- endDf[[3,3]]
percentWUS <-endDf[[2,3]]
percentCWR <-endDf[[1,3]]
#total number of genus per dataset
accessionWEP <- endDf[[3,4]]
accessionWUS <- endDf[[2,4]]
accessionCWR <- endDf[[1,4]]

```

```{r, echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}

# join the CWR data set with SOS by taxa, arrange in desc order, show only the top 20
sosCWRgenus <- trimCombinedData %>%
  filter(Type == "CWR")%>% 
  group_by(NAME) %>%
  dplyr::add_count() %>%
  select(c("NAME", "n"))%>%
  arrange(desc(n))%>%
  distinct() %>%
  dplyr::rename(CWR = "NAME", NumberOfAccession = "n")
sosCWRgenus20 <- sosCWRgenus[1:20,]

# join the WUS data set with SOS by taxa, arrange in desc order, show only the top 20
sosWUSgenus <- trimCombinedData %>% 
  filter(Type == "WUS")%>% 
  group_by(NAME) %>%
  dplyr::add_count() %>%
  select(c("NAME", "n"))%>%
  arrange(desc(n))%>%
  distinct()%>%
  dplyr::rename(WUS = "NAME", NumberOfAccession = "n")
sosWUSgenus20 <- sosWUSgenus[1:20,]


# join the WEP data set with SOS by taxa, arrange in desc order, show only the top 20
sosWEPgenus <- wepJoin %>%
  group_by(NAME) %>%
  dplyr::add_count() %>%
  select(c("NAME", "n"))%>%
  arrange(desc(n))%>%
  distinct()%>%
  dplyr::rename(WEP = "NAME", NumberOfAccession = "n")
sosWEPgenus20 <- sosWEPgenus[1:20,]


top20 <-  cbind(sosCWRgenus20, sosWUSgenus20,  sosWEPgenus20)

top20
```
The above table shows the top twenty genera collected by the SOS that are considered CWR, WUS, or WEP





```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
# 
# #Initiate an empty dataframe
# allDataSummary <- data.frame()
# 
# # create the first row of the dataframe
# sosDataTotals <- c( uniqueGen, uniqueSpe)
# 
# #filter based on the type for WUS and CWR
# wusSamples <- K13Data %>% filter(Type=="WUS")
# cwrSamples <-K13Data %>% filter(Type=="CWR")
# 
# # create K13 WEP/CWR
# K13wus <- c( n_distinct(wusSamples$Genus), n_distinct(wusSamples$Taxon))
# K13cwr <- c( n_distinct(cwrSamples$Genus), n_distinct(cwrSamples$Taxon))
# 
# ## create same for world Economic plants takes some more work
# # remove duplicates and separate based on full name into genus and species
# wepSum <- WEPData %>% distinct(Taxon) %>% separate(Taxon, c("GENUS", "SPECIES"))
# # subset data exclude all features with x as species
# wepSum1 <- filter(wepSum, SPECIES != "x")
# # preform the oppisite to pull all the avoids values out, needed a differtent method
# wepX <- filter(WEPData, grepl(" x ", Taxon))
# # seperate name then exclude xcolumn to construct a new dataframe
# wepX1 <-  wepX %>% separate(Taxon, c("GENUS", "x", "SPECIES")) %>% select(GENUS, SPECIES)
# # combine the two datasets to get a complete listing of genus and species
# WEPgs <- bind_rows(wepSum1, wepX1)
# # constuct a vector with the needed values
# WEP <- c( as.numeric(n_distinct(WEPgs$GENUS)), as.numeric(n_distinct(WEPData$Taxon)))
# 
# 
# 
# #bind the two df together
# dataSummary <- rbind(allDataSummary, sosDataTotals, K13cwr, K13wus, WEP)
# 
# #define column names and row names
# colnames(dataSummary) <- c(  "Total Number of Genera", "Total Number of Taxon")
# rownames(dataSummary) <- c("SOS", "CWR in K13", "WUS in K13",  "WEP")
# (dataSummary)
# ## I dont think I want to lose the row data by arranging it
# #arrange(dataSummary, dataSummary$`Total Number of Genus`)

```


```{r  echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE , caption="Summary of the SOS collection relative to Econmically Useful and Wild Crop Relatives.", rows.print=15}
# #Initiate an empty dataframe
# sosSummary2 <- data.frame()
# 
# # create the first row of the dataframe
# sosCollection <- c(totalSamples, sosTotalUnique, uniqueWUS, percentWUS, uniqueCWR, percentCWR, uniqueWEP, percentWEP)
# 
# #bind the two df together
# sosSummary2 <- rbind(sosSummary2, sosCollection)
# 
# #define column names and row names
# rownames(sosSummary2) <- c("SOS Collection")
# colnames(sosSummary2) <- c('Total Accessions', "Total Number of Taxa", "Taxa that are WUS per K13", "Percent of SOS Accessions that are WUS per K13", "Taxa that are CWR per K13", "Percent of SOS Accessions that are CWR per K13", "Taxa that are WEP" , "Percentage of SOS Accessions that are WEP")
# 
# sosSummary2t <- t(sosSummary2)


# create the first row of the dataframe
sosCollection <- data.frame(totalSamples, sosTotalUnique,  uniqueCWR, percentCWR, accessionCWR, uniqueWUS, percentWUS, accessionWUS, uniqueWEP, percentWEP, accessionWEP)
#bind the two df together
sosSummary2 <- t(sosCollection)

#define column names and row names
colnames(sosSummary2) <- c("SOS Collection")
labels<- t(data.frame('Total Accessions', "Total Number of Taxa",  "Taxa that are CWR per K13", "Percent of SOS Accessions that are CWR per K13", "Accessions that are CWR per K13", "Taxa that are WUS per K13", "Percent of SOS Accessions that are WUS per K13", "Accessions that are WUS per K13","Taxa that are WEP" , "Percentage of SOS Accessions that are WEP", "Accessions that are WEP"))

sosSummary2t <- bind_cols(tbl_df(labels), tbl_df(sosSummary2)) %>%
  dplyr::rename( label = "V1" )

### look up column that the unique number of taxa is being called on should be full name
### get the reference paper for WEP

(sosSummary2t)

```

The above table displays the number and proportion of SOS accessions that could be classified by Khoury et al. 2013 as CWR and/or WUS, or by Wiersema and Leon 2013 as a WEP.

```{r}
###
# specifics for CWR priority 1a 
##
cwr1a <- trimCombinedData %>%
  filter(Type == "CWR", Priority_specific2 == "Priority (1)A") %>%
  group_by(NAME) %>%
  dplyr::summarise(TotalAccession = n()) %>%
  dplyr::select(c("NAME", "TotalAccession"))%>%
  arrange(desc(TotalAccession))
cwr1a

totalP1A <- nrow(cwr1a)
```

This is a listing all all the Prioriety 1 A crop wild relatives. There are apporixmately 250 of these species in the United States. A P1A crop wild relative is a native species that is closely related to an important food crop. The SOS program has collected `r totalP1A` of theses P1A species.






## Conservation

The following table displays the diversity of SOS accessions that are currently maintained within the NPGS.


### Accessions Stored Within the ARS System


The difference in the total number of accessions collected by SOS and the number of accessions recorded in GRIN is significant. Part of this can be explained by a potential backlog of samples within the ARS and a lag in delivery time between collection and distribution. It generally takes a year from when the seeds are collect to when they are received by the ARS. There is a lagtime between collecting seed and entrance into the system due to the work of processing seed and associated data. Not all SOS accesions make it into the NPGS (why? insufficient seed?)


This graph illustrates the date that an accession was collected by the SOS program, when it was sent to ARS, and when it was backed up at NSSL. There is generally a lag time between the collection period and when ARS receives the accessions. The bar graph does show both the actual collection time and the time at which the accessions made it to the ARS. The lag effect can be seen with between the SOS(green) and ARS received (grey) in 2018. No new SOS accessions have been added but the ARS is still receiving accessions that were collected in previous years. A similar relationship can be seen between the receiving of Accessions at the ARS and the backing up of those accessions at the NLGRP (blue). While many samples were received in 2017 none of them have made it into long term back up at this point.

```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE , caption="Summary of SOS and GRIN Relative to CWR and WEP", rows.print=15}

# read in GRIN Accessions
grinData <- allGrin %>%
  mutate(NewTaxon = Taxon) %>%
  separate(NewTaxon, c("genus", "species", "other1", "Other2", "other3"))

# calcualte the total number of features and the total number of taxon
totalSamplesGRIN <- nrow(grinData)# why would these two numbers be different? Question for Renee
totalTaxon <- dplyr::n_distinct(grinData$Taxon)

# join the the GRIN data with the K13 Dataset
combinedData3 <- inner_join(grinData, K13Data, by='Taxon') 

#group by WEP and CWR and derive number of taxon and percent of sos that are those taxa
typeSum <- combinedData3 %>%
  group_by(Type) %>%
  dplyr::summarise(uniqueNumber = n_distinct(Taxon), percentType = signif(100*(n_distinct(Taxon) / sosTotalUnique),3), totalAccession = n())

#complete the process again for WEP Taxa
combinedData3a <- inner_join(grinData, wepLocal, by='Taxon') %>%
  dplyr::summarise(uniqueNumber = n_distinct(Taxon), percentType = signif(100*(n_distinct(Taxon) / sosTotalUnique),3), totalAccession = n()) %>%
  mutate(Type = "WEP")

#generate a new df with all the values
endDf1 <- bind_rows(typeSum, combinedData3a)

#calculate total number of species from each dataset
uniqueWEP1 <- endDf1[[3,2]]
uniqueWUS1 <- endDf1[[2,2]]
uniqueCWR1 <- endDf1[[1,2]]
#percent of sos in this class per dataset
percentWEP1<- endDf1[[3,3]]
percentWUS1 <-endDf1[[2,3]]
percentCWR1 <-endDf1[[1,3]]
#total number of genus per dataset
genusWEP1 <- endDf1[[3,4]]
genusWUS1 <- endDf1[[2,4]]
genusCWR1 <- endDf1[[1,4]]


# create the first row of the dataframe
grinCollection <- c(totalSamplesGRIN, totalTaxon, uniqueCWR1, percentCWR1, genusCWR1, uniqueWUS1, percentWUS1, genusWUS1, uniqueWEP1, percentWEP1, genusWEP1)

#bind the two df together
sosSummary3 <- bind_cols(sosSummary2t, tbl_df(grinCollection)) %>%
  dplyr::rename('SOS in GRIN Global' = value)
# decided against showing this table
sosSummary3
```
Not all SOS collection make into GRIN as a result the proportion of the collection that is a CWR, WUS, or WEP decrease as well. 

```{r message=FALSE, warning=FALSE, echo=FALSE}

# filter based on year collected and get a count on the number of unique accessions
sosPre2017 <- sosData %>%
  filter(COLL_YR < 2017) %>%
  nrow()

percentAtARS = signif(100*(totalSamplesGRIN/sosPre2017),4)

```





### Accession Maintained at NLGRP for Long Term Backup

Once within the NPGS, the quality of the seed collections are accessed and samples are transfered to the National Laboratory for Genetic Resource Preservation (NLGRP) in Fort Collins, Colorado for long term preservation in optimal conditions. The line graph below shows the the number of accessions maintained by the SOS, GRIN, and ARS over time.


```{r message=FALSE, warning=FALSE, echo=FALSE}
# # ### attempt at a cumulative chart

# calculate the year the Accession was received
grinDates <- grinData %>%
  separate(Received.Date, c("M", "D", "Y"))%>%
  mutate(year = as.numeric(Y))%>%
  mutate(count = as.numeric(1))

#filter for accession that are backed up
nsslDates <-grinDates %>%
  filter(Backup.Location.1 == "NSSL")

#create a count column for sos data
sosCount <- sosData %>%
  mutate(count = as.numeric(1))


sosCuml <- sosCount%>%
  group_by(COLL_YR) %>%
  dplyr::summarise(total = n())%>%
  mutate(dataset = 'SOS') %>%
  dplyr::rename(year = "COLL_YR") %>%
  mutate(cumsum = cumsum(total))


grinCuml <- grinDates %>%
  group_by(year)%>%
  dplyr::summarise(total = n())%>%
  mutate(dataset = 'GRIN')%>%
  mutate(cumsum = cumsum(total))


arsCuml <-grinDates %>%
  filter(Backup.Location.1 == "NSSL") %>%
  group_by(year)%>%
  dplyr::summarise(total = n()) %>%
  mutate(dataset = 'NLGRP')%>%
  mutate(cumsum = cumsum(total))


allCuml <- bind_rows(sosCuml, grinCuml, arsCuml)



trend <- ggplot(allCuml, aes(x=year, y=cumsum, color= dataset)) +
    labs(title = "Accumulation of Accessions Over Time ",
       x = "Collection Year",
       y= "Number of Accession Collected",
       caption = "") +
  geom_line(size=1.5)+
  theme_minimal()+
  ylim(c(0,25000))+
  xlim(c(2000,2018))+
  scale_x_continuous(breaks = seq(2000,2018, 2))

trend

```

This illustrates that the rate of accumulation between the three groups fluctuates a bit over time. The drop in number of samples incorporated into GRIN and NLGRP in the more resent years is due in part to the time lag between data collecting, seed processing, and storage. It should also be noted that individual collection must have at least 10,000 un utilized seeds in order to be sent to the USDa for long term storage and distibution. SOS collections are gather with specific resporation porjects in mind at the district level. As a result often the seed gather is actively used for revegetation purposes. This is likely the most significant factor in the difference in samples collected and samples stored.  

```{r , echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, rows.print=15}
#filter based on the "is backuped" column.
backedUp <- dplyr::filter(grinData, Is.Backed.Up. == 'Y' )
totaledBackedUp <-nrow(backedUp)

#filter based on the backup location being NLGRP
nlgrpData <- filter(backedUp, Backup.Location.1 == 'NSSL' )
totalBackedUp <- nrow(nlgrpData)


## Example of the backed up features that are missing a backup location????
#### does this mean an error in data entry, can we assume
#MissingBackup <- (backedUp$Backup.Location.1 )


#define the total number of species present in backup
totalTaxonNLGRP <- n_distinct(nlgrpData$Taxon)
totalTaxonNLGRP
# join the the GRIN data with the K13 Dataset
combinedData4 <- inner_join(nlgrpData, K13Data, by='Taxon') 

#group by WEP and CWR and derive number of taxon and percent of sos that are those taxa
typeSum4 <- combinedData4 %>%
  group_by(Type) %>%
  dplyr::summarise(uniqueNumber = n_distinct(Taxon), percentType = signif(100*(n_distinct(Taxon) / sosTotalUnique),3), totalAccession = n())


#complete the process again for Wp Taxa
combinedData4a <- inner_join(nlgrpData, wepLocal, by='Taxon')%>%
  dplyr::summarise(uniqueNumber = n_distinct(Taxon), percentType = signif(100*(n_distinct(Taxon) / sosTotalUnique),3),totalAccession = n()) %>%
  mutate(Type = "WEP")


#generate a new df with all the values
endDf2 <- bind_rows(typeSum4, combinedData4a)

#calculate total number of species from each dataset
uniqueWEP2 <- endDf2[[3,2]]
uniqueWUS2 <- endDf2[[2,2]]
uniqueCWR2 <- endDf2[[1,2]]

#percent of sos in this class per dataset
percentWEP2 <- endDf2[[3,3]]
percentWUS2 <- endDf2[[2,3]]
percentCWR2 <- endDf2[[1,3]]
#total number of genus per dataset
accessionWEP2 <- endDf2[[3,4]]
accessionWUS2 <- endDf2[[2,4]]
accessionCWR2 <- endDf2[[1,4]]

# create the first row of the dataframe
nlgrpCollection <- c(totalBackedUp, totalTaxonNLGRP, uniqueCWR2, percentCWR2, accessionCWR2, uniqueWUS2, percentWUS2, accessionWUS2,  uniqueWEP2, percentWEP2, accessionWEP2)

#bind the two df together
sosSummary4 <- bind_cols(sosSummary3, tbl_df(nlgrpCollection)) %>%
  dplyr::rename('SOS in NLGRP' = value)

sosSummary4
#define column names and row names



```

Samples at NLGRP represent the most protected accessions. Most of the data that makes into the GRIN global system makes it in the backup system at NLGRP. 



## Where are SOS accessions Maintained in the NPGS?

The challenge of coordinating the preservation of these seeds is seen in part by how my samples become orphaned within the system, lacking a curator specialized in the pertinent genus or species. Orphaned accesions are those which do not fit into any of the existing collections. Orphaned accessions continue to be held at Pullman (W6) but without a formal curator. Note the Pullman quantities in the table include both curator and non curated accessions.

```{r}
# call in sosAccessions: summaries by maintaince site or owned by ***check with Colin or Renee on this one***
# produce a table showing number of successions and proportion of total


#group by location and derive number of taxon
sosAccTaxa <- grinData %>%
  group_by(Maintenance.Site) %>%
  dplyr::summarise(uniqueNumber = n_distinct(Taxon)) %>%
  dplyr::rename(Facility = "Maintenance.Site", UniqueTaxa = "uniqueNumber" )

library(plyr)
# subet columns needed.
sosOwner <- count(grinData$Maintenance.Site)%>%
  arrange(desc(freq))%>%
  dplyr::rename(Facility = x, NumberOfAccessions = freq)

####
# need to fix this a bit
####

  # mutate(FacilityName = c("Western Regional Plant Introduction Station, Washington", "Ornamental Plant Germplasm Center, Ohio", "National Animal Disease Center, Iowa",
  #                         "Plant Introduction Research, Iowa", "NSL unknown" , "Plant Genetic Resources Conservation Unit, Georgia", "SARU unknown Plants from alaska??", "Missing Data"))

#join the two datasets
sosOwner2 <- left_join(sosOwner, sosAccTaxa, by='Facility', type='left', match='all') %>%
  dplyr::rename(FacilityAcronym = "Facility")

# %>%
#   select(Facility, FacilityAcronym, NumberOfAccessions, UniqueTaxa)

sosOwner2
```
The majority of the SOS collection is maintained by the ARS in Pullman, Washington (W6). This is where the native plant collection is based. Accession that are maintained at other locations are there due to the request of other curators. The are accessions that are very unlikely to be orphaned. It is difficult to determine from data alone how many of the accession at W6 an being actively stored but not regenerated.


## Viability of SOS seeds

Collection and storage are only part of the story. Germination and the ability to grow out the plant to collect new seeds is a large part of what the nPGS does. Often little is known about the conditions required to germinate the seeds.
```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
viability <- tbl_df(viability)
uniqueGenus <- n_distinct(viability$Genus)
percentGenus <- signif(100*(uniqueGenus/uniqueGen),4)
```

An experiment was developed and performed by Annette Miller at the NLGRP. Annette tested the viability of seeds to germinate for `r uniqueGenus` unique genera. This accounts for `r percentGenus[1,]` percent of all the genera collected by the SOS program. The data generated by this study illustrates what to expect when working with these plants in the future. The table below shows that percentage of seeds that were tested that were successfully geminated. The histogram groups those into viability categories.


```{r caption="Listing of Seed Viability by Genus", rows.print=10}
colnames(viability) <- c('Genus', 'Percent_of_Viable_Sample' )
DT::datatable(arrange(viability, desc(Percent_of_Viable_Sample)))

```
The table shows the proportion of seeds received that were alive on arrival at NLGRP. 



```{r , echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, caption="Viability of Seeds Collected by the SOS progam."}
# histogram with added parameters
hist(viability$Percent_of_Viable_Sample,
  main="Seed Lot Viability",
  xlab="Proportion of Seeds that were living",
  ylab="Number of Genera",
  col="light green",
  border="white",
  las=1, #defien the orientation of the label
  breaks=20
)

```


This distribution shows that the majority of the species collected by the SOS can be germinated given the current level of understanding. Species that had a very low germination success rate will require more research to understand what are the necessary conditions to promote successful germination. Until those methods are understood it may not be possible to maintain the population indefinitely.

## Distribution of SOS Materials from the NPGS


```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, caption="Distribution of seeds over time", rows.print=15}

##
# Eliminate the bar chart here and replace with three bar graphs of orders taxa and accession per year. 
# an accumulation per year of the line graph either one with all three or line on the bar graph  
# 
##

date <- newOrders %>%
  separate(completed_date,into=c("date", "time"), sep = ":")%>%
  separate(date, into = c("month","day", "year"), sep = "/")%>%
  subset(select = -c(time)) %>%
  filter(!is.na(year))

date$year <- substr(date$year, 0,4)

# histogram with added parameters
hist(as.numeric(date$year),
  main="Distributions of SOS accessions over time by Packet",
  xlab="Year",
  ylab="Number of Packets",
  col="light blue",
  border="white",
  ylim = c(0,2500),
  las=1 #defines the orientation of the label
)

```

This is a graph of the total number of samples delivered. Each record is assocaited with approximately 100 seeds. While this represents a good description of the total number of seeds sent the chart below splits this data to examine the unique number of order request, the number of taxons order, and the number of accessions ordered per year. 
```{r }
### need to remake a bargraph with a cumulative growth 
### this will take some work 

uniqueOrders <- date %>%
  select(year, order_request_id)%>%
  distinct() %>%
  group_by(year) %>%
  dplyr::summarise(count = n()) %>%
  mutate(use = "UniqueOrders") %>%
  dplyr::select(year,count,use)

uniqueTaxon <- date %>%
  select(year, Taxon) %>%
  distinct() %>%
  group_by(year) %>%
  dplyr::summarise(count = n()) %>%
  mutate(use = "UniqueTaxon") %>%
  select(year,count,use)


uniqueAccession <- date %>%
  select(year, accession_number) %>%
  distinct()  %>%
  group_by(year) %>%
  dplyr::summarise(count = n()) %>%
  mutate(use = "UniqueAccession") %>%
  select(year,count,use)

uniquePacket <- date %>%
  select(year) %>%
  group_by(year) %>%
  dplyr::summarise(count = n()) %>%
  mutate(use = "UniquePacket") %>%
  select(year,count,use)

allGroups <- rbind(uniqueOrders, uniqueTaxon,uniqueAccession ,uniquePacket)

ggplot(allGroups, aes(x=year,  y=count, color = use)) +
  geom_point(size=3)+
  labs(x="Year", y="Number of Features", title = 'Features Delivered Over Time') +
  scale_color_manual(labels = c("Unique Accession","Unique Orders", "Unique Packets","Unique Taxa") , values=c("#1b9e77","#d95f02","#7570b3", "#0000b3"))


```

**Working on changing this to show total growth over time rather then discrete years**
The NPGS stores and distributes to researchers accesions that collected by SOS. Seeds are distributed for a number of reasons including academic research studies and educational demonstrations. Many of the distributions take place within the ARS when seeds are sent to and from various seed labs (CK - ???). The distribution of seeds is end user driven and a result the volume of seeds transferred fluctuates over time.

```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE ,rows.print=5}

# produce the frequency of the primary intented use values 
countOrder <- plyr::count(date$intended_use_code)
# Define the undefined column type

countOrder[1,1] <- NA

arrange(countOrder, desc(freq)) %>% `colnames<-`(c("Reason for Distribution", "Individual Packets"))
```

Primary reason for distibution was incorperated into the ordering process in 2015. There are four drop down options. 

<br>
<br>


```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}

### rows are not orders replace this assumption with orders. 
ordesDF <- date[!duplicated(date$order_request_id),]
# organize data by intended us to plot 
speciesOr2 <- ordesDF %>%
  group_by(intended_use_code, year)%>%
  dplyr::summarise(number =n())

# plot data 
ggplot(speciesOr2, aes(x=year,  y=number, color = intended_use_code)) +
  geom_point(size=3)+
  labs(x="Year", y="Number of Orders",color = "Intended Use", title = 'Orders for Intended Use Over Time')+
  scale_color_manual(labels = c("NA","education","home gardening", "other", "research") , values=c("#e41a1c","#377eb8","#4daf4a","#984ea3","#ff7f00"))

```
<br>
<br>
**Working on changing this to show total growth over time rather then discrete years**
If we take a look at the deliveries over time we can see there is some distinct variability over time. The NA variable is present because before 2015 the reason for distribution was not recorded.
<br>
<br>

```{r , echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
###
# The secondary use catergories have associated drop downs with the.
# filter by primary group then secondary. Graph for both education and other.
# home gardening and other are both fill in the blank. Provide a table of rank occurance for each
###

# split the untended use notes column into secondary class and naritive 
intendedUse2 <- date %>%
  separate(intended_use_note, c("Secondary_Class", "Research_use"), "Research use notes")

intendedUse2 <- intendedUse2[!duplicated(intendedUse2$order_request_id),]

# list of all possible secondary uses for reserach 
researchUses <- c("Botanical/taxonomic investigations. ","Genetic studies. ","Entomological investigations. ","Plant Pathological investigations. ", "Chemistry. ", "Varietal Development. ", "Bioremediation. ", "Weed Science. ", "Historical, cultural and anthropological research. ")

research <- intendedUse2 %>%
  filter(intended_use_code %in% c( "RESEARCH")) %>%
  filter(Secondary_Class %in% researchUses) %>%
  group_by(Secondary_Class, year)%>%
  dplyr::summarise(number =n())

ggplot(research, aes(x=year,  y=number, color = Secondary_Class)) +
  geom_point(size=3)+
  labs(x="Year", y="Number of Orders",color = "Secondary_Class", title = 'Orders for Research Over Time')+
  scale_color_manual(labels = unique(research$Secondary_Class), values=c("#e41a1c","#377eb8","#4daf4a","#984ea3","#ff7f00","#ffff33","#a65628","#f781bf","#999999"))

```
<br>
<br>
**Working on changing this to show total growth over time rather then discrete years**
The research catergory has 9 distinct secondary classes. Their relative abundence is illistrated by the graph above.
<br>
<br>

```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
# repeat the process for educational deliveries
educationUses <- c("Public education, demonstrations. ", "Class instruction. ")

#filter for only educational uses and summaries by use and year
education <- intendedUse2 %>%
  filter(intended_use_code %in% c( "EDUCATION")) %>%
  filter(Secondary_Class %in% educationUses) %>%
  group_by(Secondary_Class, year)%>%
  dplyr::summarise(number =n()) %>%
  filter(!is.na(year))

education2 <- education %>%
  spread(Secondary_Class, year)

ggplot(education, aes(x=year,  y=number, color = Secondary_Class)) +
  geom_point(size=3)+
  labs(x="Year", y="Number of Orders",color = "Secondary_Class", title = 'Orders for Education Over Time')+
  scale_color_manual(labels = unique(education$Secondary_Class), values=c("#e41a1c","#377eb8"))
```
<br>
<br>
**Working on changing this to show total growth over time rather then discrete years**
The Education catergory has 2 distinct secondary classes. Their relative abundence is illistrated by the graph above.
<br>
<br>

The other and home gardening class do not have a standardize secondary class. Similarily all orders occur before 2015 do not have any standarized classifacation. Users requestiong orders would fill out the need for the order in a notes column on the request form. This notes column was combined with the class columns for orders after 2015. In order to pull some generalized information from this notes columb we employed some text mininng techniques to find commonly used words.


```{r,  echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, rows.print=15 }
#convert notes class to character
intendedUse2$qualData <- paste(intendedUse2$Research_use, as.character(intendedUse2$Note), sep="___")

# create a corpus??? , convert to lower case, Remove punctuation, convert to plain text and remove stopwords, preform stemming
textCorpus <- Corpus(VectorSource(intendedUse2$qualData))%>%
  tm_map(content_transformer(tolower)) %>%
  tm_map(removeWords, stopwords('english')) %>%
  tm_map( removeWords, c("will", "etc", "like", "see", "also"))%>%
  tm_map(stemDocument) #this line is a touch one... It changes alot of the words and I had to manuel correct some I probably missed a few

dtm <- TermDocumentMatrix(textCorpus)
m <- as.matrix(dtm)
v <- sort(rowSums(m),decreasing=TRUE)
d <- data.frame(word = names(v),freq=v)

commonWords <- d %>%
  filter(freq > 30)


#### do not rerun
##write.csv(commonWords, "commonWords2018.csv")
## I complete multiple manual alterations to this document in Excel. could pull it off in R
####

# read in the clean CSV of work frequency
trimList <- read.csv("commonWords1.csv") %>%
  filter(process %in% c(0,1)) %>%
  arrange(desc(Number.of.Occurances)) %>%
  select(-c("X"))# removes undescroptive common words

DT::datatable(trimList)
```
The table above illustate all words that have occur over 30 times in the written notes of the orders. Notes are not required for orders and while it can be assumed that the number of times a word appears is positively related to the number of orders it appear in not efforts were made to not count words repeated used in a single order.
The full list of words contians information about what was to be done with the order and details about the order itself( what species). To distiguish between the tool we accessed if the word was associated with a process or not. In the table above 1 indicates the world is related to a process, 0 does not. We generated specific word clounds based on these two groups.  
```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, rows.print=15}

set.seed(1234)
# generate a word cloud for all words relatibe to process
processWords <- trimList %>%
  filter(process == 1) %>%
  select(c("Word","Number.of.Occurances"))
colnames(processWords) <- c("Word", "Number of Occurances")

wordcloud(words = processWords$Word, freq = processWords$`Number of Occurances`, min.freq = 1,
          max.words=150, random.order=FALSE, rot.per=0.35,
          colors=brewer.pal(8, "Dark2"))
```
<br>
<br>

 word cloud displaying all the process related words


 <br>
 <br>

```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, rows.print=15}

# generate a word cloud for all words not related to a process
otherWords <- trimList %>%
  filter(process == 0) %>%
  select(c("Word","Number.of.Occurances"))
colnames(otherWords) <- c("Word", "Number of Occurances")

wordcloud(words = otherWords$Word, freq = otherWords$`Number of Occurances`, min.freq = 1,
          max.words=150, random.order=FALSE, rot.per=0.35,
          colors=brewer.pal(8, "Paired"))

```
<br>
<br>
A word cloud displaying all non process related words.  
<br>
<br>


```{r  echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, rows.print=5  }
#print a table of all Orders rancked by primary, second order type and containing merge notes layers

nonDupUse <- intendedUse2[!duplicated(intendedUse2$order_request_id),]

allText <- nonDupUse %>%
  select("intended_use_code", "Secondary_Class","qualData" )
DT::datatable(allText)
```
<br>
<br>
This table contains a complete listing of all the user inputs regarding what they attended to use the seeds for. 



```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}

# reproducing with new dataset 
genus <-date %>%
  mutate(taxa = Taxon) %>%
  tidyr::separate(taxa, into=c("genus_name", "species", "var_SSP", "subUnit"))


genusSum <- genus %>%
  group_by(genus_name)%>%
  dplyr::summarise(genusSum = n())%>%
  arrange(desc(genusSum))%>%
  dplyr::rename(Genus = genus_name, "Individual Samples Sent" = genusSum)


#calculate the total number of seeds sent
totalSeedsSent <- sum(genusSum$`Individual Samples Sent`)

```


<br>
<br>

```{r, echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}

K13Data$genus_name <- K13Data$Genus

#Call in the WUP/CWR data, summarize by Genus, join to distribution data by genus
K13genus <- K13Data %>%
  group_by(genus_name) %>%
  summarise_()%>%
  inner_join(genus,by="genus_name")


numberGenus <- n_distinct(K13genus$genus_name)

# calculated the total number of seeds sent
totalWUSSet <- sum(K13genus$COUNT)


#Repeat for WUS features
WEPgenus <- wepLocal %>%
  separate(Taxon, c("genus_name", "species","other1", "other2")) %>%
  group_by(genus_name) %>%
  summarise_()%>%
  inner_join(genus,by="genus_name")

numberGenusWEP <- n_distinct(WEPgenus$genus_name)


# calculated the total number of seeds sent
totalWEPSet <- sum(WEPgenus$COUNT)

#define the unique number of genus and seeds sent to enitity groups
reciever <- genus %>%
  group_by(To.Whom) %>%
  dplyr::summarise(numberOfGenus = n_distinct(genus_name), numberOfSeeds = n()) %>%
  arrange(desc(numberOfGenus))%>%
  dplyr::rename("Entity Receiving Seeds" = 'To.Whom'  ,"Number Of Genera Received" = 'numberOfGenus', "Samples Distributed" = 'numberOfSeeds')

DT::datatable(reciever)

```
<br>
<br>
**working on the Entiry receiving seeds, not sure how specic it should be** 
This table shows the locations that are receiving the most SOS seeds. The number of unique genera delivered and the number of packets distributed are both shown. 



**incorrect numbers at the moment, please ignore for now**
There are a total of `r totalSeedsSent` samples that have been distributed by the NPGS. Of that `r totalWUSSet`, are considered to be either CWR or WUS and `r totalWEPSet` are consider to be WEP. This is representing `r numberGenus`  and `r numberGenusWEP` unique genera respectively.
<br>
<br>


<br>
<br>

```{r,echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
#totalSent -- not super relavent because it is the number of time indiviudal species were sent
totalSent <- date %>%
  group_by(Taxon) %>%
  dplyr::summarise(sum = n(), accession = n_distinct(accession_number)) %>%
  arrange(desc(sum)) %>%
  `colnames<-`(c("Taxon", "Total Packets Delivered", "Total Accessions Pulled From"))

DT::datatable(totalSent)
```
<br>
<br>

This chart shows the number of times a specific taxon  has been delivered. These deliveries could include multiple accessions.
<br>
<br>
```{r, echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE}
# who is receiving deliveries
requestor <- date %>%
  group_by(category_code) %>%
  filter(!is.na(quantity_shipped))%>%
  dplyr::summarise(taxa = n_distinct(Taxon), count = n_distinct(order_request_id), sum = n()) %>%
  arrange(desc(sum))%>%
  `colnames<-`(c("Category Code", "Total Unique Taxon Received", "Total Orders", "Total Packets"))
  

DT::datatable(requestor)

```

This graphic shows which user group is receiving the greatest diversity of taxa.




# Discussion

- there is a lot of room for narrative and exploration of these results. Collaboration with individuals more involved in the processing of these accession need to be pulled in to provide opinions and highlight significance of results per each section


# Extras (just ideas really)
The SOS data contains a wealth of information regarding the environment surrounding the collection sites. Depending on the interest and direction of this document, there are alot of interesting routes this data can lead. 

One idea we tossed around a bit it showcasing a specific species in more detail. For example Heliantheis Annus is the most collected P1A CWR. This could entail a map of collection site or a showcase of the plants that are present around this species. 


```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, rows.print=15}
### exploration of detail account of a single species. Need a map here 
sosFull <- read.csv(paste0(base_path, "/analysisData/fullSOSreport_20180614.csv"))

heli <- sosFull %>%
  filter(NAME == "Helianthus annuus") %>%
  mutate(phyto = PHYTOREGION_FULL)%>%
  separate(phyto,c("phytoShort","phytoExtra"), sep ="(Omernik)" )
```

The table show all the species associated with the the collection site. 


```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, rows.print=15}

### building frame work for a network analysis between geographic region and plant associations 
heli2 <- heli %>%
  dplyr::select(PHYTOREGION_FULL, ASSOCIATED_TAXA_FULL) %>%
  separate(ASSOCIATED_TAXA_FULL, c("sp1","sp2","sp3","sp4","sp5","sp6","sp7","sp8","sp9","sp10","sp11","sp12","sp13","sp14","sp15","sp16"), sep=":") %>%
  arrange(PHYTOREGION_FULL)

DT::datatable(heli2)
```

The table shows all the species associated with the the collection site. It can be filtered to show varied phytoregion. I think this is very interesting for understanding potential genetic diversity across the landscape.  


```{r echo=FALSE, tidy=TRUE, message=FALSE, warning=FALSE, rows.print=15}
library(gdalUtils)
library(tmap)
library(rgdal)
library(knitr)
library(tm)
library(sp)
xy <- heli %>% 
  mutate(x=as.numeric(as.character(heli$LONGITUDE_DECIMAL)), 
         y = as.numeric(as.character(heli$LATITUDE_DECIMAL)))%>%
  dplyr::select(x,y)
  

spatialData <- sp::SpatialPointsDataFrame(coords = xy, data = heli , proj4string = CRS("+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs"))

tmap_mode("view")
qtm(shp = spatialData,dots.col = "phytoShort" , title = "Helianthesis Annus Collects by Phytoregion")
```
**use arrows to move around the map... A little odd sorry** 
Very basic map, but could offer some opportunity to highlight collection within a specific regions (sagebrush habitat), Recent burned areas. 
Again just lots and lots of possabilities. 





*work on adding the reference into the text when appropriate.*

@maxted_towards_2006

@khoury_inventory_2013

@harlan_genetic_1976

@noauthor_flora_nodate

@commission_on_genetic_resources_for_food_and_agriculture_second_2010


# References

Found out how to make this work!!!
You need to add the intext reference by @ + reference call from the bib file.
If it is cited within the paper the full reference will be added to the doc. What's nice about this is that you can pull from a bit .bib file and only reference specific papers from it.
